File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p02572/p02572_num8_parsed.cpp
#include <iostream>
#include <vector>

using namespace std;

const int MOD = 1000000007;

int main() {
    int N;
    cin >> N;
    vector<int> A(N);
    for (int i = 0; i < N; i++) {
        cin >> A[i];
    }
    
    long long sum = 0;
    for (int i = 0; i < N; i++) {
        for (int j = i + 1; j < N; j++) {
            sum = (sum + (1ll * A[i] * A[j]) % MOD) % MOD;
        }
    }
    
    cout << sum << endl;
    
    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p02572/p02572_num8_parsed.py
# Program to find the sum of A_i * A_j over all pairs (i,j) such that 1 <= i < j <= N, modulo (10^9+7)

# Function to calculate the sum of A_i * A_j
def find_sum_of_products(N, arr):
    MOD = 10**9 + 7
    sum = 0
    
    # Loop to calculate the sum of A_i * A_j
    for i in range(N-1):
        for j in range(i+1, N):
            sum += arr[i] * arr[j]
            sum %= MOD
    
    return sum

# Reading input from standard input
N = int(input())
arr = list(map(int, input().split()))

# Calling the function to find sum of A_i * A_j 
result = find_sum_of_products(N, arr)

# Printing the result
print(result)